src/matrices.jl

```diff
@@ -0,0 +1,53 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENCE in the project root.
+# ------------------------------------------------------------------
+
+abstract type CoDaMatrix{T} end
+
+"""
+    JMatrix{T}
+
+Square matrix of ones.
+"""
+struct JMatrix{T} <: CoDaMatrix{T} end
+JMatrix{T}(d::Integer) where {T} = Ones{T}(d, d)
```

Comment:
Do you think we can erase this constructor from the type directly? From what I understand we will only need to pass the dimension d to an instance of the type:

```julia
J(d) # where J is an instance of JMatrix
```

The same comment applies to all other matrices.